### PR TITLE
Gt/make diffwave compo

### PR DESCRIPTION
**Pull request by gregtucker**
Yet another overland flow component, but one that fills a niche: a diffusion-wave representation that linearizes the friction law, so that flow is treated as a diffusion problem with a space- and time-varying diffusivity. An advantage of the component is that it preserves the role of pressure gradients in spreading flow (which kinematic-wave models lack), but omits the complexity associated with the momentum terms in the shallow-water equations. It uses a simple forward-Euler explicit solution scheme, but with adaptive time-stepping that uses small sub-steps when needed for stability. It is designed for use in geomorphic problems where only a relatively simply approximation of the flow field is needed, but where the lateral spreading of flow is also important.

### CODE REVIEW

**Comment by mcflugen** [comment]
@gregtucker This looks wonderful! I'll try to give it a closer look of this afternoon.

File: landlab/components/overland_flow/linear_diffusion_overland_flow_router.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        infilt_rate : float, optional (defaults to 0)
+            Maximum rate of infiltration, m/s
+        velocity_scale : float, defaults to 1
+            Characteristic flow velocity, m/s
+        cfl_factor : float, optional (defaults to 0.2)
+            Time-step control factor: fraction of maximum estimated time-step
+            that is actually used (must be <=1)
+        """
+        super().__init__(grid)
+
```
[the remark this answers is not in the document]
Comment (mcflugen) [reply landlab/components/overland_flow/linear_diffusion_overland_flow_router.py:166]: And for `infilt_depth_scale ` just to eliminate possible division by zero errors,
```python
if infilt_depth_scale <= 0.0:
    raise ValueError(f"infilt_depth_scale must be greater than zero {infilt_depth_scale}")
```

**Review by mcflugen — COMMENTED** [review]
Looks good @gregtucker! I've made only a few suggestions.

I'm not sure of the best way to suggest small changes to the notebook. Is it ok if I just make them myself and push them to this branch? They're fairly trivial.

I assume you'll add tests. Don't forget to add the new component to the docs.

**Comment by gregtucker** [comment]
The test failure should be fixed now; the notebook failure is unrelated to this component, and seems to come from the known issue of OpenTopography now requiring keys to access their data.

**Comment by gregtucker** [comment]
> 

Yes of course, please add away!

**Comment by mcflugen** [comment]
@BCampforts Have you had a chance to look at this?

File: landlab/components/overland_flow/linear_diffusion_overland_flow_router.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        self._cfl_param = cfl_factor * 0.5 * np.amin(grid.length_of_link) ** 2
+
+    @property
+    def rain_rate(self):
+        """Rainfall rate"""
+        return self._rain
+
+    @rain_rate.setter
+    def rain_rate(self, value):
```
[the remark this answers is not in the document]
Comment (gregtucker) [reply landlab/components/overland_flow/linear_diffusion_overland_flow_router.py:194]: Done, and even figured out how to test it via try...except...else (lmk if you know a better way)

File: landlab/components/overland_flow/linear_diffusion_overland_flow_router.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        """Velocity coefficient.
+
+        (1/(roughness^2 x velocity_scale)
+        """
+        return self._vel_coef
+
+    def _cfl_time_step(self):
+        """Calculate maximum time-step size using CFL criterion for explicit
+        FTCS diffusion."""
+        max_water_depth = max(np.amax(self._depth), _MICRO_DEPTH)
```
[the remark this answers is not in the document]
Comment (gregtucker) [reply landlab/components/overland_flow/linear_diffusion_overland_flow_router.py:208]: Nice! Trick added

File: landlab/components/overland_flow/linear_diffusion_overland_flow_router.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+        infilt = self._infilt * (1.0 - np.exp(-self._depth / self._infilt_depth_scale))
+
+        # Rate of change of water depth
+        dHdt = self._rain - infilt - dqda
+
+        # Update water depth: simple forward Euler scheme
+        self._depth[self._grid.core_nodes] += dHdt[self._grid.core_nodes] * iter_dt
+
+        # Very crude numerical hack: prevent negative water depth (TODO: better)
+        self._depth[np.where(self._depth < 0.0)[0]] = 0.0
```
[the remark this answers is not in the document]
Comment (gregtucker) [reply landlab/components/overland_flow/linear_diffusion_overland_flow_router.py:250]: Replaced

**Pull request merged**